app/pages/patientdata/patientdata.js

```diff
@@ -567,19 +612,18 @@ let mapStateToProps = state => ({
 let mapDispatchToProps = dispatch => bindActionCreators({
   fetchPatient: actions.async.fetchPatient,
   fetchPatientData: actions.async.fetchPatientData,
+  clearPatientData: actions.sync.clearPatientData,
   fetchMessageThread: actions.async.fetchMessageThread,
-  updateLocalPatientData: actions.sync.updateLocalPatientData,
   closeMessageThread: actions.sync.closeMessageThread,
 }, dispatch);
 
 let mergeProps = (stateProps, dispatchProps, ownProps) => {
   var api = ownProps.routes[0].api;
-  return _.merge({}, ownProps, stateProps, dispatchProps, {
+  return Object.assign({}, ownProps, stateProps, dispatchProps, {
     fetchers: getFetchers(dispatchProps, ownProps, api),
     uploadUrl: api.getUploadUrl(),
     onRefresh: dispatchProps.fetchPatientData.bind(null, api),
```

Comment:
Maybe this is behind the refresh not working? `onRefresh` is still calling the old `fetchPatientData` with the embedded data processing if I'm reading this right. Don't you want to map it to the async action creator `fetchPatientData` from the `fetchers`?